src/Symfony/Component/Dsn/Tests/Factory/MemcachedFactoryTest.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Tests\Adapter;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Dsn\Factory\MemcachedFactory;
+
+/**
+ * @requires extension memcached
+ */
+class MemcachedFactoryTest extends TestCase
+{
+    public function testOptions()
+    {
+        $client = MemcachedFactory::create(array(), array(
+            'libketama_compatible' => false,
+            'distribution' => 'modula',
+            'compression' => true,
+            'serializer' => 'php',
+            'hash' => 'md5',
+        ));
+
+        $this->assertSame(\Memcached::SERIALIZER_PHP, $client->getOption(\Memcached::OPT_SERIALIZER));
+        $this->assertSame(\Memcached::HASH_MD5, $client->getOption(\Memcached::OPT_HASH));
+        $this->assertTrue($client->getOption(\Memcached::OPT_COMPRESSION));
+        if (version_compare(phpversion('memcached'), '2.2.0', '>=')) {
+            $this->assertSame(0, $client->getOption(\Memcached::OPT_LIBKETAMA_COMPATIBLE));
+        }
+        $this->assertSame(\Memcached::DISTRIBUTION_MODULA, $client->getOption(\Memcached::OPT_DISTRIBUTION));
+    }
+
+    /**
+     * @dataProvider provideBadOptions
+     * @expectedException \ErrorException
+     * @expectedExceptionMessage constant(): Couldn't find constant Memcached::
+     */
+    public function testBadOptions($name, $value)
+    {
+        MemcachedFactory::create(array(), array($name => $value));
+    }
+
+    public function provideBadOptions()
+    {
+        return array(
```

Comment:
In the other providers of this test we use `yield`. Could we use it here too for consistency?

Reply:
done